### PR TITLE
Add a :quiet option to disable final output of certain classes

### PR DESCRIPTION
**Pull request by slowbro**
### Describe the change

Hi Piotr, loving the libraries! I wanted a way to make a menu system using `Prompt#select` without it leaving the 'breadcrumb trail' of all the previous selections. For example, before:

```
$ # https://gist.github.com/slowbro/c4c1fbb9f972117079a117ea5d19dc17
$ ruby ./example.rb 
Really Cool Menu Coolness Status
Coolness Status Back
Really Cool Menu Manage Coolness
Coolness is at 0.
Manage Add coolness
Coolness is at 1.
Manage Add coolness
Coolness is at 2.
Manage 
  1. Add coolness
  2. Back
_ 3. Exit

```

It's a little hard to see what's going on (and all the terminal recording software I tried did not like all the line-clearing..) - but every time a line-item is selected, a new 'breadcrumb' is added - i.e. at the root, "Really Cool Menu", upon selecting "Coolness Status", a line is left behind: "Really Cool Menu Coolness Status"

Compare that to the version with `:quiet`:

```
$ # https://gist.github.com/slowbro/8517a8e04c41719edcee821f823d351c
$ ruby ./example-quiet.rb                                                                                                                                                                            
Coolness is at 2.
Manage 
  1. Add coolness
  2. Back
_ 3. Exit
```

I did the same exact keystrokes - into status, back, into manage, add twice. But the output is a lot cleaner!

This also applies to a number of other classes:

- slider
- question (and derivatives)
- enum_list
- expander
- mask
- yes

I hope this is a desirable change!

Other minor changes:
- Added vim modeline to lib/*
- Updated some confusing/broken docs
- misc formatting fixes

### Why are we doing this?
I really want to be able to have a clean menu system using tty-prompt :)

### Benefits
Broader customization will expand usage possibility.

### Drawbacks
None that I can see - It's a new feature that should not overlap with anything.

### Requirements
Put an X between brackets on each line if you have done the item:
[x] Tests written & passing locally?
[x] Code style checked?
[x] Rebased with `master` branch?
[x] Documentaion updated?

### CODE REVIEW

File: lib/tty/prompt/question.rb [hunk 1]
```diff
@@ -251,6 +253,13 @@ def default?
         @default != UndefinedSetting
       end
 
+      # Set quiet mode.
+      #
+      # @api public
+      def quiet(value)
```
[the remark this answers is not in the document]
Comment (slowbro) [reply lib/tty/prompt/question.rb]: seems to be part of the original author's style

**Comment by slowbro** [comment]
@piotrmurach Wanted to make sure this was seen :) If there is any changes you'd like to see made, please let me know.

**Comment by piotrmurach** [comment]
@slowbro Hey, I haven't forgotten, promise ;-) I got a bit busy traveling to the USA to give a talk at a conference, I flew like 5 times in the past 2 weeks, super exhausted. I wrote down a few thoughts based on my initial review but I will need to give this another look once I'm back.

**Comment by slowbro** [comment]
i'm closing this to un-muddle my 'master'; i'll re-open with a new, updated branch.

**Comment by piotrmurach** [comment]
I can see the usefulness in having this feature. One of my reservations about this was the choice of configuration option name whether it was descriptive/informative enough and clear for people to understand what it does. So if I recall correctly, the idea is to skip printing the result after selection so that you can print over it with a new prompt. So maybe option should be `display_result: false`? Anyhow, I'm open to this feature and would like to see it as part of the library, including an example of usage in the examples folder.

**Comment by slowbro** [comment]
Agreed, the change of name would be a good idea. I'll change that & add an example before I open a new PR :)

**Comment by piotrmurach** [comment]
@slowbro I picked what I could from this PR and included the `:quiet` option in the main project as it is a very useful feature.  So thank you! ❤️

**Pull request closed without merging**